packages/v4-sdk/src/functions/bin/getBinPool.ts

```diff
@@ -0,0 +1,70 @@
+import { BigintIsh, Currency } from '@pancakeswap/swap-sdk-core'
+import invariant from 'tiny-invariant'
+import { Address } from 'viem'
+import { MAX_BIN_STEP, MIN_BIN_STEP } from '../../constants'
+import { TEN_PERCENT_FEE } from '../../constants/fee'
+import type { BinTree } from '../../types'
+import { getSortedCurrencies } from '../../utils/getSortedCurrencies'
+
+type ActiveId = number | `${number}`
+type Reserve = {
+  reserve0: bigint
+  reserve1: bigint
+}
+
+export type BinPoolState = {
+  currency0: Currency
+  currency1: Currency
```

Comment:
Let's use `currencyX` and `currencyY` to keep the naming aligned with smart contract